File: KuWanX/app/src/main/java/com/tch/kuwanx/ui/BaseActivity.java
package com.tch.kuwanx.ui;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Window;
import com.tch.kuwanx.utils.StatusBarCompat;

import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;

/**
 * Created by Jiaop on 2017/10/23.
 * activity基类
 */

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //去除标题栏
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
        //设置沉浸式状态栏
        StatusBarCompat.compat(this, Color.parseColor("#FCCB05"));
    }

    /**
     * 设置字体
     *
     * @param newBase
     */
    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
    }

}

File: KotlinPro/jiaoplibs_kotlin/src/main/java/com/jiaop/kotlin/libs/base/BaseActivity.kt
package com.jiaop.kotlin.libs.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

/**
 * Created by jiaop
 * Activity基类
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //隐藏标题栏
        supportActionBar?.hide()


    }


}